script/generate_sorted_docids_mapping.py

```diff
@@ -0,0 +1,36 @@
+ #!/usr/bin/env python
+
+import string
+from argparse import ArgumentParser
+
+def dump(output, sorted_docids):
+    fout = file(output, "w")
+    for newid in range(0, len(sorted_docids)):
+        # Note that sorted_docids[newid] is the currid.
+        fout.write(str(sorted_docids[newid]) + " " + str(newid) + "\n")
+
+
+def main():
+    parser = ArgumentParser(description='Take a document lexicon (e.g. \'.documents\' or\
+                            \'.urls\' file) and sort it, generating a file mapping \'<current ID>\
+                            <new ID>\' to use with the \'suffle_docids\' script.')
+    parser.add_argument('documents', help='File containing one document (or url) per line and \
+                        where each line number (starting from zero) represents its docid')
+    parser.add_argument('output', help='Output file mapping \'<current ID> <new ID>\'')
+    args = parser.parse_args()
+
+    # Documents dict, where each element maps an id with a document (or url).
+    documents = dict()
+    docid = 0
+    for line in open(args.documents, "r"):
```

Comment:
We don't seem to be using `docid` outside of the loop, so let's use `enumerate`:
```python
for docid, line in enumerate(open(args.documents, "r")):
    documents[docid] = string.lower(line[:-1])
```

Moreover, we can do:

```python
with open(args.documents, "r") as input_file:
    documents = {docid: string.lower(line[:-1]) for docid, line in enumerate(input_file)}
```